src/streamlit_app.py

```diff
@@ -298,13 +301,34 @@ def push_assistant_query_result(
         self,
         df: typing.Any,
     ) -> typing.Dict[str, typing.Any]:
+        key = str(uuid.uuid4())[:8]
         msg = {
             "type": HistoryItemTypes.QUERY_RESULT.value,
             "role": Roles.ASSISTANT.value,
             "data": df,
+            "key": key,
             "text": "",
         }
 
+        if 0 < len(df) < _MAX_VALUES_TO_USE_FOR_FOLLOWUP_QUESTION:
```

Comment:
would you want to say "no data" to the llm when none is found? so it can make conclusions based on that

Reply:
I can send an empty json, with the columns and everything (basically - remove the check condition of greater than 0) - wdyt?

Reply:
probably fine

Reply:
Fixed